ultrasonic/src/main/kotlin/org/moire/ultrasonic/fragment/AlbumRowAdapter.kt

```
@@ -87,4 +103,34 @@ class AlbumRowAdapter(
     override fun newViewHolder(view: View): RecyclerView.ViewHolder {
         return ViewHolder(view)
     }
+
+    /**
+     * Handles the star / unstar action for an album
+     */
+    @Suppress("TooGenericExceptionCaught")
+    private fun onStarClick(entry: MusicDirectory.Entry, star: ImageView) {
+        entry.starred = !entry.starred
+        star.setImageDrawable(if (entry.starred) starDrawable else starHollowDrawable)
+        val musicService = getMusicService()
+        Thread {
+            val useId3 = shouldUseId3Tags
+            try {
+                if (entry.starred) {
+                    musicService.star(
+                        if (!useId3) entry.id else null,
+                        if (useId3) entry.id else null,
+                        null
+                    )
+                } else {
+                    musicService.unstar(
+                        if (!useId3) entry.id else null,
+                        if (useId3) entry.id else null,
+                        null
+                    )
+                }
+            } catch (exception: Exception) {
+                Timber.e(exception)
```

Little tip:
If you write  `catch (all: Exception)`, then you won't need to suppress an error ;)

---

Oh thanks! I'll try to remember that :)